src/services/embeddings_service.py

```
@@ -0,0 +1,278 @@
+        }
+
+        # Check API key
+        if self.config.provider == "openai":
+            results["api_key_configured"] = bool(os.getenv("OPENAI_API_KEY"))
+
+        # Test embedding generation
+        if self._client and results["api_key_configured"]:
+            try:
+                test_embedding = await self.create_embedding("test")
+                results["test_embedding_success"] = len(test_embedding) > 0
+            except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)